### PR TITLE
proc,service,terminal: eval expressions in the scope of a deferred call

### PR DESCRIPTION
**Pull request by aarzilli**
```
proc,service,terminal: eval expressions in the scope of a deferred call

Add ability to evaluate variables on the scope of a deferred call's
argument frame.

```

### CODE REVIEW

**Comment by derekparker** [comment]
Needs rebase.

File: pkg/terminal/command.go [hunk 1]
```diff
@@ -700,6 +706,22 @@ func (c *Commands) frameCommand(t *Term, ctx callContext, argstr string, directi
...
+
+	space := strings.Index(argstr, " ")
+
+	var err error
+	ctx.Scope.DeferredCall, err = strconv.Atoi(argstr[:space])
+	if err != nil {
+		return err
+	}
+	if ctx.Scope.DeferredCall <= 0 {
+		return errors.New("argument of deferred must be a number greater than 1")
```
[the remark this answers is not in the document]
Comment (aarzilli) [reply pkg/terminal/command.go]: > and a few other flags are listed in the documentation that do not work and aren't parsed
Weird.

**Comment by aarzilli** [comment]
Done.

File: Documentation/cli/README.md [hunk 2]
```diff
@@ -339,10 +348,11 @@ If regex is specified only the source files matching it will be returned.
 ## stack
 Print stack trace.
 
-	[goroutine <n>] [frame <m>] stack [<depth>] [-full] [-g] [-s] [-offsets]
+	[goroutine <n>] [frame <m>] stack [<depth>] [-full] [-offsets] [-defer]
 
 	-full		every stackframe is decorated with the value of its local variables and arguments.
-	-offsets	prints frame offset of each frame
+	-offsets	prints frame offset of each frame.
+	-defer	prints calls deferred by each frame.
```
Comment (derekparker) [inline Documentation/cli/README.md]: nit: let's reword this a bit, something like: `prints deferred function call stack for each frame`.

Comment (aarzilli) [reply Documentation/cli/README.md]: Done.

**Pull request merged**